lib/common/http1client.c

```diff
@@ -560,51 +560,57 @@ static h2o_iovec_t build_request(struct st_h2o_http1client_t *client, h2o_iovec_
 static void on_connection_ready(struct st_h2o_http1client_t *client)
 {
     h2o_iovec_t proxy_protocol = h2o_iovec_init(NULL, 0);
-    int chunked = 0;
     h2o_iovec_t connection_header = h2o_iovec_init(NULL, 0);
     h2o_httpclient_properties_t props = {
-        &proxy_protocol,
-        &chunked,
-        &connection_header,
+        &proxy_protocol, &connection_header
     };
     h2o_iovec_t method;
     h2o_url_t url;
     h2o_header_t *headers;
     size_t num_headers;
-    h2o_iovec_t body;
+    h2o_httpclient_req_body_t body = {.streaming.content_length = SIZE_MAX};
```

Comment:
I think that the initialization should be `{H2O_HTTPCLIENT_REQ_BODY_NONE}`.

The problem of specifying `.stream.content_length` is that it's semantically incorrect, because the type would not be `H2O_HTTPCLIENT_REQ_BODY_STREAMING`. Note that `streaming` is within an union that is interpreted differently based on the value of `type`.